llvm/lib/Target/RISCV/GISel/RISCVCallLowering.cpp

```diff
@@ -303,6 +307,9 @@ static bool isSupportedArgumentType(Type *T, const RISCVSubtarget &Subtarget) {
     return true;
   if (T->isPointerTy())
     return true;
+  // TODO: Support fixed vector types.
+  if (T->isVectorTy() && T->isScalableTy() && Subtarget.hasVInstructions())
```

Comment:
This function is also used by `RISCVCallLowering::lowerCall`, but I don't see any tests for that.

Reply:
Thanks for catching this. As the PR is titled, I only intended on implementing lowerFormalArguments. I pushed a fixup below to fix this for now. The changes by the fixup should be removed when lowerCall support is added.

I don't think lowerCall should block this change since once this change goes in, it unblocks instruction selection for functions that take vector arguments, just not calls with vector arguments.

Reply:
Don't you need support for returning vectors too to keep the instructions from being trivially dead?

Reply:
Implementing vector store could do the same thing. I was planning on doing lowerReturn right after this patch. Do you want me to add it to this one?  I still think this patch unblocks work on GISel for vectors at O0 without returns and stores.

Reply:
Returns should be a separate patch. I was trying to understand your overall plans.